backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/BPartnerBankAccountImportHelper.java

```diff
@@ -93,8 +100,101 @@ else if (!Check.isEmpty(importRecord.getSwiftCode(), true) && !Check.isEmpty(imp
+			}
+
+			bankAccount.setIBAN(importRecord.getIBAN());
+		}
+
+		if (Check.isNotBlank(importRecord.getQR_IBAN()))
+		{
+			if (bankAccount == null)
+			{
+				final BankAccountQuery queryWithQRIban = BankAccountQuery.builder().qrIban(importRecord.getQR_IBAN()).build();
+				bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithQRIban));
+			}
+
+			bankAccount.setQR_IBAN(importRecord.getQR_IBAN());
+		}
+
+		if (bankAccount == null)
+		{
+			return;
+		}
+
+		if (bankAccount.getC_Currency_ID() <= 0 && Check.isBlank(importRecord.getISO_Code()))
+		{
+			throw new AdempiereException("Missing mandatory value for currency on business partner bank account!");
+		}
+
+		final CurrencyCode currencyCode = CurrencyCode.ofThreeLetterCode(importRecord.getISO_Code());
+		final CurrencyId currencyId = currencyBL.getByCurrencyCode(currencyCode).getId();
+		bankAccount.setC_Currency_ID(currencyId.getRepoId());
+
+		bankAccount.setC_BPartner_ID(bpartnerId.getRepoId());
+		bankAccount.setAD_Org_ID(importRecord.getAD_Org_ID());
+
+		InterfaceWrapperHelper.saveRecord(bankAccount);
+
+		importRecord.setC_BP_BankAccount_ID(bankAccount.getC_BP_BankAccount_ID());
+	}
+
+	@NonNull
+	private I_C_BP_BankAccount initializeBankAccountFromBankDetails(@NonNull final String bankDetails)
+	{
+		final I_C_BP_BankAccount bankAccount;
+
+		final Optional<BankId> bankIdOpt = resolveBankIdFromBankDetails(bankDetails);
+
+		if (bankIdOpt.isPresent())
+		{
+			final BankAccountQuery queryWithBankId = BankAccountQuery.builder().description(bankDetails).build();
+			bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithBankId));
+			bankAccount.setC_Bank_ID(BankId.optionalToRepoId(bankIdOpt));
+		}
+		else
+		{
+			final BankAccountQuery queryWithBankDetails = BankAccountQuery.builder().description(bankDetails).build();
+			bankAccount = handleBPBankAccountCollection(bankAccountDAO.getBpartnerBankAccount(queryWithBankDetails));
+
+			if (Check.isBlank(bankAccount.getName()))
+			{
+				bankAccount.setName(bankDetails);
+			}
+
+			if (Check.isBlank(bankAccount.getDescription()))
+			{
+				bankAccount.setDescription(bankDetails);
+			}
+		}
 
 		return bankAccount;
 	}
 
+	@NonNull
+	private Optional<BankId> resolveBankIdFromBankDetails(@NonNull final String bankDetails)
+	{
+		final Set<BankId> bankIdCollection = bankRepository.retrieveBankIdsByName(bankDetails);
+
+		return Optional.ofNullable(CollectionUtils.singleElementOrNull(bankIdCollection));
+	}
+
+	@NonNull
+	private I_C_BP_BankAccount handleBPBankAccountCollection(final List<BPartnerBankAccount> bpBankAccountCollection)
+	{
+		return Optional.ofNullable(CollectionUtils.singleElementOrNull(bpBankAccountCollection))
+				.map(BPartnerBankAccount::getId)
+				.map(id -> InterfaceWrapperHelper.load(id, I_C_BP_BankAccount.class))
+				.orElseGet(() -> InterfaceWrapperHelper.newInstance(I_C_BP_BankAccount.class));
+	}
```

Comment:
- I'd refactor to `getSingleBPBankAccount(): Optional<I_C_BP_BankAccount>` and keep the method only about loading the I_C_BP_BankAccount form the `List<BPartnerBankAccount>`
  - I'd actually remove this method all together and add a new method in `bankAccountDAO`
- another thing, we should not use `InterfaceWrapperHelper` to `load` or `create a new instance` outside of DAO/Repo classes, so pls go via `BankAccountDAO`